### PR TITLE
Make rendering PRF stage optional

### LINKED ISSUE #106: Provide mechanism to show/hide PRF stage identifier for different usage
The PRF stage abbreviation is usage dependent:

* In rendering a PRF document, the PRF document type is ignored (i.e. pretend to be "ISO 6709:2022"). 
* In referencing a PRF document, the PRF document type is rendered (i.e. "ISO/PRF 6709:2022").

This means we need to have some option of specifying behavior here.

> Before: `ISO 1000`. After: `ISO PRF 1000`
> 
> * We are not ignoring stage PRF in identifier rendering.

_Originally posted by @ronaldtse in https://github.com/metanorma/pubid-iso/issues/102#issuecomment-1251774037_

### PR DESCRIPTION
**Pull request by mico**
closes #106

### CODE REVIEW

File: lib/pubid/iso/identifier.rb [hunk 1]
```diff
@@ -97,17 +97,17 @@ def urn
     def formatted(format)
       case format
       when :ref_num_short
-        to_s(with_language_code: :single, stage_format_long: false)
+        to_s(with_language_code: :single, stage_format_long: false, with_prf: true)
```
Comment (ronaldtse) [inline lib/pubid/iso/identifier.rb:100]: @mico can we also allow `with_prf:` in `formatted`, or better yet, allow the `formatted` options inside `to_s`?

**Review by ronaldtse — CHANGES_REQUESTED** [review]

File: lib/pubid/iso/identifier.rb (hunk 1 above)
Comment (mico) [reply lib/pubid/iso/identifier.rb]: > @mico can we also allow `with_prf:` in `formatted`, or better yet, allow the `formatted` options inside `to_s`?

Could you please check if that is what you mean?

Comment (ronaldtse) [reply lib/pubid/iso/identifier.rb]: Can we do:
```
pubid.to_s(format: :ref_num_short)
# and
pubid.to_s(format: :ref_num_short, with_prf: true)
```
?

**Review by ronaldtse — APPROVED** [review]
Excellent changes -- thanks!

**Pull request merged**